ote_sdk/ote_sdk/utils/tiler.py

```diff
@@ -7,51 +7,197 @@
 
+from ote_sdk.utils.detection_utils import detection2array
+from ote_sdk.utils.nms import multiclass_nms
+
 
 class Tiler:
     """Tile Image into (non)overlapping Patches. Images are tiled in order to efficiently process large images.
 
     Args:
         tile_size: Tile dimension for each patch
         overlap: Overlap between adjacent tile
-        batch_size: Batch Size
+        max_number: max number of prediction per image
+        segm: enable instance segmentation mask output
     """
 
-    def __init__(self, tile_size: int, overlap: float, batch_size=1) -> None:
+    def __init__(
+        self,
+        tile_size: int,
+        overlap: float,
+        max_number: int,
+        model: Any,
+        segm: bool = False,
+    ) -> None:
         self.tile_size = tile_size
         self.overlap = overlap
         self.stride = int(tile_size * (1 - overlap))
-        self.batch_size = batch_size
+        self.max_number = max_number
+        self.model = model
+        self.segm = segm
 
-    def tile(self, image: np.ndarray) -> np.ndarray:
+    def tile(self, image: np.ndarray) -> List[List[int]]:
         """Tiles an input image to either overlapping, non-overlapping or random patches.
 
         Args:
             image: Input image to tile.
 
         Returns:
-            Tiles generated from the image.
+            Tiles coordinates
         """
         height, width = image.shape[:2]
 
-        tiles, offsets = [], []
-        for (loc_i, loc_j) in product(
-            range(0, height - self.tile_size + 1, self.stride),
+        coords = [[0, 0, width, height]]
+        for (loc_j, loc_i) in product(
             range(0, width - self.tile_size + 1, self.stride),
+            range(0, height - self.tile_size + 1, self.stride),
         ):
-            tiles.append(
-                image[
-                    loc_i : (loc_i + self.tile_size),
-                    loc_j : (loc_j + self.tile_size),
-                    ...,
+            coords.append(
+                [loc_j, loc_i, loc_j + self.tile_size, loc_i + self.tile_size]
+            )
+        return coords
+
+    def predict(self, image: np.ndarray):
+        """Predict by cropping full image to tiles
+
+        Args:
+            image (np.ndarray): full size image
+            segm (bool, optional): return mask if enabled. Defaults to False.
+
+        Returns:
+            detection: prediction results
+            features: saliency map and feature vector
+        """
+        detections = np.empty((0, 6), dtype=np.float32)
+        features = [None, None]
+        masks: List[np.ndarray] = []
+        for i, coord in enumerate(self.tile(image)):
+            feats, output = self.predict_tile(image, coord, masks, i == 0)
+            detections = np.append(detections, output, axis=0)
+            # cache full image feature vector and saliency map at 0 index
+            if i == 0:
+                features = feats
+
+        if np.prod(detections.shape):
```

Comment:
AFAIK, `np.prod(x)` returns 1 when `x=np.array([])` . Is it okay? How about using `.shape` if you want to check whether `scores` is empty or not

Reply:
It's not `np.prod(detections)` but `np.prod(detections.shape)` so it's fine